dipy/core/sphere.py

```diff
@@ -496,6 +497,44 @@ def disperse_charges(hemi, iters, const=.2):
     return HemiSphere(xyz=charges), potential
 
 
+def fibonacci_sphere(n_points=724, randomize=True):
+    """
+    Generate points on the surface of a sphere using Fibonacci Spiral Sampling.
+
+    Parameters
+    ----------
+
+    - n_points (int): The number of points to generate on the sphere surface.
+      Default is 724.
+    - randomize (bool): If True, randomize the starting point on the sphere.
+      Default is True.
+
+    Returns
+    -------
+
+    array-like (N,3): An array of 3D points representing coordinates
+    on the sphere surface.
+    """
+    random_shift = 0
+    if randomize:
+        random_generator = np.random.default_rng()
+        random_shift = random_generator.integers(0, n_points)
+
+    points = []
+    offset = 2.0 / n_points
+    increment = np.pi * (3. - np.sqrt(5.))
+
+    for i in range(n_points):
+        y = ((i * offset) - 1) + (offset / 2)
+        r = np.sqrt(1 - pow(y, 2))
+        phi = ((i + random_shift) % n_points) * increment
+        x = np.cos(phi) * r
+        z = np.sin(phi) * r
+        points.append([x, y, z])
```

Comment:
this can be done with numpy array and it will be way faster !  I do not see the need of a loop here. so this fucntion can be done without loop. 

Please, can you update ? thanks

Reply:
Now the array is pre populated and updates each point. Thanks for the suggestion